src/expand.iced

```diff
@@ -0,0 +1,96 @@
+parse3 = require './parse3'
+pgp_utils = require('pgp-utils')
+{katch,json_stringify_sorted} = pgp_utils.util
+triplesec = require('triplesec')
+{ExpansionError} = require('./errors').errors
+{prng,createHmac} = require 'crypto'
+
+#========================================================
+
+check_expansions = ({expansions}) ->
+  for k,v of expansions
+    check_expansion_kv {k, v}
+
+#========================================================
+
+exports.hmac_obj = hmac_obj = ({obj, key}) ->
+  hmac = createHmac("sha256", key)
+  s = json_stringify_sorted obj
+  hmac.update(Buffer.from(s, 'utf8')).digest()
+
+#========================================================
+
+check_expansion_kv = ({k,v}) ->
+  if not parse3.is_hex(k, 32) then throw new ExpansionError "bad hash: #{k}"
+  if not parse3.is_hex(v.key, 16) then throw new ExpansionError "bad hmac key: #{v[1]}"
```

Comment:
we're going with 16 then? (only bringing this up again since these aren't short lived)

Reply:
i think it's fine....